### DISCUSSION TITLE
람다 표현식과 익명 클래스의 차이

### DISCUSSION THREAD

**Discussion by BvrPark** [Study-2-Modern-Java-In-Action/Study-2-Modern-Java-in-Action#33] [CHAPTER 3 람다 표현식 - Q&A]
* 람다 표현식으로 표현하면, 클래스를 작성하고 객체를 생성하지 않아도 메서드를 사용할 수 있다. 
* 익명 클래스는 이름이 없는 클래스로, 클래스 선언과 동시에 해당 클래스의 인스턴스를 생성한다. 

</br>

이처럼 람다와 익명 클래스는 둘 다 객체를 생성하지 않고도 메서드를 사용할 수 있다.
그렇다면 둘의 문법적인 차이점이 아닌 기능적인 차이점에는 어떤 것들이 있을까?

**Comment by corock** [comment]
제가 Q&A 올린 글과 소개하고자 하는 내용이 중복되는 것 같아 저는 패스하겠습니다~!

**Comment by JoisFe** [comment]
`익명 클래스보다는 람다를` 유명한 책인 `effective java`의 item의 제목 중 하나입니다.
그 이유는 익명 클래스는 가독성이 매우 떨어지나 람다를 활용하면 간결해져 가독성이 좋아집니다.
가독성이 좋아지면 그만큼 유지보수에도 좋은 것 같습니다.
(다만 람다도 어지럽게 쓰면 가독성이 쓰읍..)
